src/pages/DownloadMonitor.vue

```diff
@@ -19,18 +19,45 @@
-                                <p>{{Math.min(Math.floor(downloadObject.progress), 100)}}% complete</p>
-                                <Progress v-if="!downloadObject.failed"
-                                    :max='100'
-                                    :value='downloadObject.progress'
-                                    :className="['is-info']"
-                                />
-                                <Progress v-else-if="downloadObject.failed"
-                                    :max='100'
-                                    :value='100'
-                                    :className="['is-danger']"
-                                />
+
+                                <div class="row" v-if="!downloadObject.failed">
+
+                                    <div class="col">
+                                        <p v-if="downloadObject.downloadProgress < 100">Downloading: {{ downloadObject.modName }}</p>
+                                        <p v-else-if="downloadObject.installProgress < 100 && !downloadObject.failed">Downloading:</p>
+                                        <p>{{Math.min(Math.floor(downloadObject.downloadProgress), 100)}}% complete</p>
+                                        <Progress
+                                            :max='100'
+                                            :value='downloadObject.downloadProgress'
+                                            :className="['is-info']"
+                                        />
+                                    </div>
+
+                                    <div v-if="downloadObject.installProgress < 100 && !downloadObject.failed" class="col">
+                                        <p v-if="downloadObject.installProgress < 100">Installing: {{ downloadObject.modName }}</p>
+                                        <p v-else>Installing:</p>
+                                        <p>{{Math.min(Math.floor(downloadObject.installProgress), 100)}}% complete</p>
+                                        <Progress
+                                            :max='100'
+                                            :value='downloadObject.installProgress'
+                                            :className="['is-info']"
+                                        />
+                                    </div>
+
+                                </div>
+
+                                <div v-else-if="downloadObject.failed">
+                                    <p>{{downloadObject.failed ? "Download failed" : `${Math.min(Math.floor(downloadObject.installProgress), 100)}% complete`}}</p>
+                                    <Progress
+                                        :max='100'
+                                        :value='100'
+                                        :className="['is-danger']"
+                                    />
+                                </div>
+
+                                <div v-else>
```

Comment:
Is this ever executed? `downloadObject.failed` is boolean, so one of the previous `if`s is bound to execute. Or it can be `undefined`, in which case `v-if="!downloadObject.failed"` will execute. And IMO it makes no sense to spefically check against undefined, as showing "status unknown" looks odd when we can't event show `downloadObject.initialMods` above goes assumably it's also `undefined`.

Reply:
This one made me laugh when I read what I had coded LOL